datasets/spectra.py

```diff
@@ -0,0 +1,55 @@
+from astropy.coordinates import SkyCoord, Angle
+from gammapy.datasets import gammapy_extra
+from gammapy.image import ExclusionMask
+from gammapy.obs import DataStore
+from gammapy.region import SkyCircleRegion
+from gammapy.spectrum import SpectrumAnalysis
+from gammapy.utils.energy import EnergyBounds
+
+center = SkyCoord(83.63, 22.01, unit='deg', frame='icrs')
+radius = Angle('0.3 deg')
+on_region = SkyCircleRegion(pos=center, radius=radius)
+
+bkg_method = dict(type='reflected')
+
+#J ai defini la variable GAMMAPY_EXTRA dans mon .bashrc qui pointe sur le repertoire gammapy-extra
+#exclusion_file = gammapy_extra.filename("test_datasets/spectrum/"
+#                                        "dummy_exclusion.fits")
+exclusion_file ="/Users/jouvin/Desktop/these/test_Gammapy/gammapy-extra/datasets/exclusion_masks/tevcat_exclusion.fits"
+excl = ExclusionMask.from_fits(exclusion_file)
+
+bounds = EnergyBounds.equal_log_spacing(1, 10, 40, unit='TeV')
+
+#store = gammapy_extra.filename("datasets/hess-crab4")
+#ds = DataStore.from_dir(store)
+obs = [23523, 23559, 23592, 23526]
+ds = DataStore.from_dir('hess-crab4')
+ana = SpectrumAnalysis(datastore=ds, obs=obs, on_region=on_region,
```

Comment:
When I run the code spectra.py on the four Crab observations in gammapy-extra I got this error when I define a SpectrumAnalysis object.
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/Users/jouvin/Desktop/these/test_Gammapy/gammapy-extra/datasets/spectra.py in <module>()
     26 ds = DataStore.from_dir('hess-crab4')
     27 ana = SpectrumAnalysis(datastore=ds, obs=obs, on_region=on_region,
---> 28                        bkg_method=bkg_method, exclusion=excl, ebounds=bounds)
     29 
     30 ana.write_ogip_data(outdir='ogip_data')

/Users/jouvin/gammapy/gammapy/spectrum/spectrum_analysis.py in __init__(self, datastore, obs, on_region, exclusion, bkg_method, nobs, ebounds)
    107             raise ValueError("No valid observations found")
    108         if bkg_method['type'] == 'reflected':
--> 109             mask = self.filter_by_reflected_regions(bkg_method['n_min'])
    110             self._observations = self.observations[mask]
    111 

KeyError: u'n_min'
```

When I remove the from __future__ import absolute_import, division, print_function, unicode_literals from the class spectrumanalysis I still get the same error for "n_min":
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/Users/jouvin/Desktop/these/test_Gammapy/gammapy-extra/datasets/spectra.py in <module>()
     26 ds = DataStore.from_dir('hess-crab4')
     27 ana = SpectrumAnalysis(datastore=ds, obs=obs, on_region=on_region,
---> 28                        bkg_method=bkg_method, exclusion=excl, ebounds=bounds)
     29 
     30 ana.write_ogip_data(outdir='ogip_data')

/Users/jouvin/gammapy/gammapy/spectrum/spectrum_analysis.py in __init__(self, datastore, obs, on_region, exclusion, bkg_method, nobs, ebounds)
    107             raise ValueError("No valid observations found")
    108         if bkg_method['type'] == 'reflected':
--> 109             mask = self.filter_by_reflected_regions(bkg_method['n_min'])
    110             self._observations = self.observations[mask]
    111 

KeyError: 'n_min'
```